core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/tuple/Tuple.java

```diff
@@ -199,17 +203,25 @@ private static void checkAttributeMatchesField(Attribute attribute, Object field
      */
     public static class Builder {
 
+        private final Schema schema;
         private final Schema.Builder schemaBuilder;
         private final HashMap<String, Object> fieldNameMap;
 
         /**
          * Creates a new Tuple Builder.
          */
         public Builder() {
+            this.schema = null;
             this.schemaBuilder = new Schema.Builder();
             this.fieldNameMap = new HashMap<>();
         }
 
+        public Builder(Schema schema) {
+            this.schema = schema;
```

Comment:
These two constructors can get confusing. Are you planning to remove one of them later?

Reply:
Yes, that's part of my proposal that I want to validate, before going ahead, as I'll have to change all operators. I've written a description of my proposal in the PR.